typeagent/knowpro/interfaces.py

```diff
@@ -1,911 +1,82 @@
-        max_matches: int | None = None,
-        threshold_score: float | None = None,
-    ) -> list[ScoredMessageOrdinal]: ...
-
-    async def lookup_messages_in_subset(
-        self,
-        message_text: str,
-        ordinals_to_search: list[MessageOrdinal],
-        max_matches: int | None = None,
-        threshold_score: float | None = None,
-    ) -> list[ScoredMessageOrdinal]: ...
-
-    # Async alternatives to __len__ and __bool__
-    async def size(self) -> int: ...
-
-    async def is_empty(self) -> bool: ...
-
-    # TODO: Others?
-
-    async def serialize(self) -> "MessageTextIndexData": ...
-
-    async def deserialize(self, data: "MessageTextIndexData") -> None: ...
-
-
-class IConversationSecondaryIndexes[TMessage: IMessage](Protocol):
-    property_to_semantic_ref_index: IPropertyToSemanticRefIndex | None
-    timestamp_index: ITimestampToTextRangeIndex | None
-    term_to_related_terms_index: ITermToRelatedTermsIndex | None
-    threads: IConversationThreads | None = None
-    message_index: IMessageTextIndex[TMessage] | None = None
-
-
-class IConversation[
-    TMessage: IMessage,
-    TTermToSemanticRefIndex: ITermToSemanticRefIndex,
-](Protocol):
-    name_tag: str
-    tags: list[str]
-    messages: "IMessageCollection[TMessage]"
-    semantic_refs: "ISemanticRefCollection"
-    semantic_ref_index: TTermToSemanticRefIndex
-    secondary_indexes: IConversationSecondaryIndexes[TMessage] | None
-
-
-# -------------
-# Search Types
-# -------------
-
-
-@dataclass
-class SearchTerm:
-    """Represents a term being searched for.
-
-    Attributes:
-        term: The term being searched for.
-        related_terms: Additional terms related to the term. These can be supplied
-            from synonym tables and so on.
-            - An empty list indicates no related matches for this term.
-            - `None` indicates that the search processor may try to resolve related
-              terms from any available secondary indexes (e.g., ITermToRelatedTermsIndex).
-    """
-
-    term: Term
-    related_terms: list[Term] | None = CamelCaseField(
-        "Additional terms related to the term. These can be supplied from synonym tables and so on",
-        default=None,
-    )
-
-
-# Well-known knowledge properties.
-type KnowledgePropertyName = Literal[
-    "name",  # the name of an entity
-    "type",  # the type of an entity
-    "verb",  # the verb of an action
-    "subject",  # the subject of an action
-    "object",  # the object of an action
-    "indirectObject",  # the indirect object of an action
-    "tag",  # tag
-    "topic",  # topic
+"""Aggregated knowpro interfaces for backwards compatibility."""
+
+from __future__ import annotations
+
+from .interfaces_core import *  # noqa: F401,F403
+from .interfaces_indexes import *  # noqa: F401,F403
+from .interfaces_search import *  # noqa: F401,F403
+from .interfaces_serialization import *  # noqa: F401,F403
+from .interfaces_storage import *  # noqa: F401,F403
```

Comment:
What tool requires the `# noqa` comments? Not any of the tools we're currently using in the repo or CI, I believe.

Or is this part of switching to ruff instead of black? I'm not yet convinced that we need that. It certainly seems we'd need to think about that separately. (It does appear to find a lot of redundant imports and variable assignments.)